src/transforms/remap.rs

```diff
@@ -133,14 +135,52 @@ pub struct RemapConfig {
     #[configurable(derived, metadata(docs::hidden))]
     #[serde(default)]
     pub runtime: VrlRuntime,
+
+    #[configurable(derived, metadata(docs::hidden))]
+    #[serde(skip)]
+    #[derivative(Debug = "ignore")]
+    /// Cache can't be `BTreeMap` or `HashMap` because of `TableRegistry`, which doesn't allow us to inspect tables inside it.
+    /// And even if we allowed the inspection, the tables can be huge, resulting in a long comparison or hash computation
```

Comment:
I'm not clear how this part is relevant. When scanning the table, it is necessary to do an equality comparison on each element, so we are already taking this hit. The bigger issue would be if we can actually implement `Hash` on the key to use `HashMap`, or `PartialOrd` to use `BTreeMap`.

Reply:
We will take this hit every time we need to check a cache. 

>  The bigger issue would be if we can actually implement Hash on the key to use HashMap, or PartialOrd to use BTreeMap.
Sorry, I don't understand that part. Should I specify why exactly we can't implement `Hash` or `PartialEq`? Honestly I don't think it'll ever have a lot of records.

Reply:
With second line I've tried to say that using `HashMap` or `BTreeMap` requires deeper inspection and it can be slow, because `Vec` uses only shallow comparison. I'll clarify it in comment

Reply:
I think this is still a bit confusing. I agree with Bruce here that this is   "The bigger issue would be if we can actually implement Hash on the key to use HashMap, or PartialOrd to use BTreeMap.".